setup.py

```diff
@@ -87,6 +87,7 @@
     "z3-solver>=4.7",
 ]
 bip_requirements = ["cplex", "docplex"]
+csp_requirements = ["python-constraint>=1.4"]
```

Comment:
How likely is it that a downstream library would want to depend on `CSPLayout`?  If it's plausible, then we might want to add an extra with no content, wait a release, and only then move `python-constraint` to the extras.  That way libraries will be able to depend on `qiskit-terra[constraint]` without that requiring only the latest version of Terra.